cmd/internal/ecs/add.go

```diff
@@ -0,0 +1,330 @@
+package ecs
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/AlecAivazis/survey/v2/terminal"
+	"github.com/akitasoftware/akita-cli/cmd/internal/cmderr"
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
+	"golang.org/x/term"
+)
+
+// Helper function for reporting telemetry
+func reportStep(stepName string) {
+	telemetry.WorkflowStep("Add to ECS", stepName)
+}
+
+// A function which executes the next part of the workflow,
+// and picks a next state or exits. Return "true" to continue
+// so that the zero value exits.
+type AddWorkflowState func(*AddWorkflow) (cont bool, err error)
```

Comment:
This sounded totally reasonable, but:

```
cmd/internal/ecs/add.go:93:9: cannot use optionals.Some(getProfileState) (value of type optionals.Optional[func(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowState], err error)]) as type optionals.Optional[AddWorkflowState] in return statement
```

This only works if either the type is an alias, or every function literal is wrapped in a AddWorkflowState constructor.  :(

It might be better to forgo the Optional here and use nil as the end value, which I don't love.

Reply:
Added some helpers to make this work less painfully.